Homework/2/Program.cs

```diff
@@ -0,0 +1,470 @@
+                    break;
+                case "3":
+                    ListUsers();
+                    break;
+                case "4":
+                    PromptAddUserToArray(addToEnd: false, addToStart: true);
+                    break;
+                case "5":
+                    PromptAddUserToArray();
+                    break;
+                case "6":
+                    PromptAddUserToArray(addToEnd: false, addToStart: false);
+                    break;
+                case "7":
+                    PromptRemoveUser(removeFromEnd: false, removeFromStart: true);
+                    break;
+                case "8":
+                    PromptRemoveUser(removeFromEnd: true, removeFromStart: false);
+                    break;
+                case "9":
+                    PromptRemoveUser(removeFromEnd: false, removeFromStart: false);
+                    break;
+                case "10":
+                    SortUsers();
+                    break;
+                case "11":
+                    LogOut();
+                    break;
+                case "12":
+                    WriteUsersToFile();
+                    Environment.Exit(0);
+                    break;
+                default:
+                    MenuSelection();
+                    break;
+            }
+        }
+
+        static string PromptInput(string message) {
+            Console.WriteLine(message);
+            return Console.ReadLine();
+        }
+
+        //Login function - gets username/password and checks against array
+        static void LogIn() {
+            Console.Clear();
+            string username = PromptInput("Enter username:");
+            string password = PromptInput("Enter password:");
+
+            if (CheckIfUserExists(username))
+            {
+                string[] userDetails = ReturnUserByUsername(username);
+                Console.WriteLine(userDetails[0] + " " + userDetails[1]);
+                if (userDetails[1] == password) {
+                    loggedIn = true;
+                    MenuSelection();
+                    return;
+                }
+            }
+
+            if (RetryConfirmation("Bad username or password."))
+            {
+                LogIn();
+            }
+            else {
+                MenuSelection();
+            }
+        }
+
+        static void LogOut() {
+            Console.Clear();
+            loggedIn = false;
+            MenuSelection();
+        }
+
+        // Function to display error message and return true for retry;
+        static bool RetryConfirmation(string messageToDisplay) {
+            Console.Clear();
+            Console.WriteLine(messageToDisplay+" Retry? y/n");
+
+            if (Console.ReadLine().ToLower() == "y")
+            {
+                return true;
+            }
+            return false;
+        }
+
+
+        // Returns true if username is found
+        static bool CheckIfUserExists(string username) {
```

Comment:
This method and `ReturnUserByUsername` are very similar.  If you changed `ReturnUserByUsername` to return `null` if it didn't find anything then this method could become...

```
static bool CheckIfUserExists(string username) {
  return ReturnUserByUsername(username) != null;
}
```